config/base.py

```diff
@@ -14,6 +14,15 @@
 DATA_ROOT = Path(os.getenv('DATA_ROOT', HERE / '_db'))
 
 
+def _getenv(var_name, default):
```

Comment:
This function is a nice simple wrapper around `os.getenv(variable_name)`, but how does it differ functionally from using simply `os.getenv(variable_name, default_value)`?  If we are going to use this function, let's add a `default=None` to this function's signature and upgrade _all_ instances within `config/base.py` to use `_getenv`.  We could also likely upgrade the usage in a handful of config / boilerplate files in the project:

- `tasks/__init__.py`
- `integration_tests/conftest.py`
- `config/codex.py`
- `config/__init__.py`
- `config/mws.py`
- `config/utils.py`
- `config/base.py`
- `tests/conftest.py`
- `scripts/docs/generate_models_graph.py`

Reply:
The difference is with `os.getenv(variable_name, default_value)`, the `default_value` is only used if `variable_name` is not in the environment, but `_getenv(variable_name, default_value)` returns `default_value` if `variable_name` is not in the environment or if `variable_name` is empty.  If we rely on being able to set an environment variable to an empty string then we can't use `_getenv`

Reply:
This is equivalent to what `os.getenv` is doing:

```python
if variable_name in os.environ:
    return os.environ[variable_name]
return default_value
```

This is equivalent to what `_getenv` is doing:

```python
if os.environ.get(variable_name):
    return os.environ[variable_name]
return default_value
```

So for example, if I have a environment variable `RECAPTCHA_PUBLIC_KEY` and it's an empty string, `os.getenv('RECAPTCHA_PUBLIC_KEY', 'xxx')` returns `""`, `_getenv('RECAPTCHA_PUBLIC_KEY', 'xxx')` returns `"xxx"`.

Are you suggesting we add a flag to `_getenv` to support both cases?

Reply:
Yes, I think it is handy to make it clear which envars allow us to accept empty fields

Reply:
Thank you for making this change.  I very much prefer the explicit flag for `empty_ok`.

Reply:
:+1: I think it's good too